crypto.go

```diff
@@ -74,12 +79,17 @@ func (s dhkemScheme) getEphemeralKeyPair(rand io.Reader) (KEMPrivateKey, KEMPubl
 		return s.skE, s.skE.PublicKey(), nil
 	}
 
-	return s.group.GenerateKeyPair(rand)
+	ikm := make([]byte, s.PrivateKeySize())
+	rand.Read(ikm)
+
+	return s.group.DeriveKeyPair(ikm)
 }
 
 func (s dhkemScheme) extractAndExpand(dh []byte, kemContext []byte, Nzz int) []byte {
-	prk := s.KDF.LabeledExtract(nil, "dh", dh)
-	return s.KDF.LabeledExpand(prk, "prk", kemContext, Nzz)
+	idBuffer := make([]byte, 2)
+	binary.BigEndian.PutUint16(idBuffer, uint16(s.ID()))
+	prk := s.group.internalKDF().LabeledExtract(nil, string(idBuffer)+"prk", dh)
```

Comment:
Shouldn't the extract label be `string(idBuffer)+"eae_prk"`, based on https://github.com/cfrg/draft-irtf-cfrg-hpke/pull/122?

Reply:
Yeah, that PR just changed this morning. I'll update this to match soon. :)

Reply:
Aha! No rush. I was just giving my code a second pass vs the spec. I'm still learning the GitHub UI, apparently 🙃

Reply:
Hehe, you and me both.

Reply:
@dmcardle updated!